### ISSUE TITLE
Disallow or enforce spaces inside of curly braces in objects?

### ISSUE THREAD

**Issue by deep-c** [Glavin001/atom-beautify#802]
A lot of the linting configs i have tried use a 1 char space at the beginning and end within objects. Would be great if we were able to get this simple setting.

I use eslint with eslint-config-airbnb and this is the most frequent issue I have when attempting to beatify my code.

http://eslint.org/docs/rules/object-curly-spacing

Thanks

**Comment by prettydiff** [comment]
Pretty Diff has an option for this: prettydiff.com/documentation.xhtml#bracepadding

It seems the option has been poorly maintained and is currently broken though.  I have already written a fix and it will be published with the next patch: https://github.com/prettydiff/prettydiff/issues/266

Currently, it seems this Pretty Diff option is not supported by Atom Beautify, so I need it add it in.  There is a JS Beautify option for brace-style which takes a value of `expand`, which is supported by Atom Beautify.  It should get you what you need.

**Comment by deep-c** [comment]
Hi thanks for the quick reply. What is the turn around for patches like this both on your end and for it to be supported in atom-beautify?

**Comment by prettydiff** [comment]
I typically publish resolutions to defect between an hour and two weeks depending upon the defect and what other items are in process.  Atom Beautify publishes changes a bit more slowly because it has a wider distribution of dependencies and potential problem states to consider.

**Comment by djbobbydrake** [comment]
+1

**Comment by prettydiff** [comment]
Launched in v1.16.15 of Pretty Diff 24 days ago.

**Comment by cortopy** [comment]
@prettydiff Brace style options do not seem to have any effect with Pretty Diff. Was this fixed?

**Comment by prettydiff** [comment]
Should be fixed.  The comment above says it was fixed with version 1.16.15 and Atom Beautify uses version 1.16.17.

**Comment by cortopy** [comment]
In that case I think there has been a regression either in Pretty Diff or Atom Beautify. Using Atom Beautify 0.29.11 and Pretty Diff, Brace Style settings have no effect at all

I've tried using "bracepadding" in `.jsbeautifyrc` but I don't think that setting is ever parsed

**Comment by Glavin001** [comment]
Correct, `bracepadding` option is not yet supported. Waiting on PR: https://github.com/Glavin001/atom-beautify/pull/1193
See https://github.com/Glavin001/atom-beautify/issues/1191#issuecomment-246546211 for some more info on this. #524 has a list of unsupported options, although there are likely more now.

**Comment by ElegantSudo** [comment]
What's going on with this?

**Comment by Glavin001** [comment]
Pull Request #1193 was merged. Published to v0.29.12

**Comment by ElegantSudo** [comment]
@Glavin001 how would I use this in my config file? I'm not seeing anything in the docs about `bracepadding` property.

**Comment by ElegantSudo** [comment]
@Glavin001 let me know when you get the chance! :)

**Comment by Glavin001** [comment]
@ElegantSudo It is named `space_in_paren`. See under https://github.com/Glavin001/atom-beautify/blob/master/docs/options.md#pretty-diff for details. You can see `space_in_paren` at the bottom of the merged Pull Request changes: https://github.com/Glavin001/atom-beautify/pull/1193/files and https://github.com/qwemaze/atom-beautify/blob/32a6fce1a6588612e82e6525b8a495041baa0e5f/src/beautifiers/prettydiff.coffee#L44

**Comment by ElegantSudo** [comment]
@Glavin001 Thanks, but I'm talking about things like this: `import { example } from './example'`. So basically just curly brackets, not parentheses.

**Comment by spo** [comment]
Any updates on this? Thanks

**Comment by yurikilian** [comment]
waiting for this option too

**Comment by Glavin001** [comment]
For those waiting for this option:

1) What beautifier are you using? Likely one of the following:
- JS-Beautify: https://github.com/beautify-web/js-beautify
- Pretty Diff: https://github.com/prettydiff/prettydiff/
2) Ensure the beautifier has the option you want
- If not, go to the respective GitHub Issues page and create an Issue! Provide a link back to here for other users to follow.
3) Ensure the option is available on Atom-Beautify
  - For the given language (JavaScript?), check supported options:
    - https://github.com/Glavin001/atom-beautify/blob/master/src/languages/javascript.coffee#L25
  - For the given beautifier, check supported options:
    - Pretty Diff: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L10-L44
    - https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/js-beautify.coffee#L13
  - If missing, submit a Pull Request to Atom-Beautify. Feel free to ask questions!

**Comment by derwaldgeist** [comment]
I got the `space_in_paren` option working now. This turns imports like
```js
import {Dimensions} from 'react-native';
```
into
```js
import { Dimensions } from 'react-native';
```
Great. However, it also applies spaces to regular parens, like function calls:
```js
Math.floor( height * 0.3 ),
```
and even arrow functions:
```js
const test = ( ) => {}
```
which eslint doesn't like. Is there a chance to configure beautification in a way that spaces only apply to object literals?

**Comment by Glavin001** [comment]
> If you're developing React with eslint and atom-beautify, it's pretty hard to keep both in sync and in sync with the numerous code samples you can find on the web. It would be awesome if there was a kind of recommended configuration file setup for this.

This idea was brought up with https://github.com/Glavin001/atom-beautify/issues/286 and has since moved to https://github.com/Glavin001/atom-beautify/issues/299 . I plan on developing https://github.com/Unibeautify/unibeautify-options-importer to solve this problem by adapting technologies like `eslint` with Unibeautify's (formerly Atom-Beautify) configuration format.

**Comment by derwaldgeist** [comment]
Sounds great!

**Comment by jordangarside** [comment]
Just to confirm:
There is no way to add spaces just between brackets atm?

**Comment by Glavin001** [comment]
@jordangarside You can take a look at available Javascript (and other language) options here: https://github.com/Glavin001/atom-beautify/blob/master/docs/options.md#javascript

Looks like the closest options: `space_in_paren` 

Check out the docs for Unibeautify (rewrite of Atom-Beautify): https://unibeautify.com/docs/option-object-curly-spacing.html#javascript
This may be what you are looking for. Let me know what you think! 😃

**Comment by jordangarside** [comment]
@Glavin001 the `space_in_paren` has the problem that it adds spaces to parens and brackets.

That Unibeautify option looks like what I want. :+1:
I'll give it a try soon.

Thanks!

**Comment by Glavin001** [comment]
Unfortunately, Unibeautify is not released yet. @szeck87 is making great strides with https://github.com/Glavin001/atom-beautify/pull/1935 . I recommend you subscribe to https://github.com/Glavin001/atom-beautify/issues/1174 for updates to Atom-Beautify.

Also, please star and/or watch https://github.com/Unibeautify/unibeautify for updates for Unibeautify. Thank you for your interest!!

**Comment by Glavin001** [comment]
If I am not mistaken, the original goal of this issue was to add support for [`object-curly-spacing`](https://eslint.org/docs/rules/object-curly-spacing) rule from ESLint.

Pretty Diff has a `bracepadding` option however this does not give desired results.
I am going to reopen this issue and welcome Pull Requests adding support for specifically `object_curly_spacing`. Please let me know if you have any questions!

**Comment by jordangarside** [comment]
Thanks for the info @Glavin001 !

For the time being I just ended up going with [Prettier](https://prettier.io/).

**Comment by Glavin001** [comment]
@szeck87 is working on adding Prettier support to Atom-Beautify as part of https://github.com/Glavin001/atom-beautify/issues/1497
